docs/inheritance_and_composition.md

```diff
@@ -0,0 +1,36 @@
+# Template Inheritance and Composition
+
+## What is Inheritance, what is Composition?
+
+Template Inheritance is a way to create a new template based on an existing one. It
+allows you to reuse code and structure from the parent template while adding or
+overriding specific parts in the child template. This is useful for creating variations
```

Comment:
I think it's mostly about _extensions_, not so much _variations_ – perhaps a matter of definition. Variations – in my mind – can often be addressed by conditional questions and conditional rendering in the same template. WDYT?

```suggestion
overriding specific parts in the child template. This is useful for creating extensions
```